android-lib/src/androidTest/java/com/kinvey/androidTest/store/DeltaCacheTest.java

```diff
@@ -0,0 +1,461 @@
+        assertTrue(store.isDeltaSetCachingEnabled());
+        testManager.cleanBackend(store, StoreType.SYNC);
+
+        Person person = new Person();
+        person.setUsername("DeltaCacheUserName");
+        person.set("CustomField", "CustomValue");
+        Person savedPerson = store.save(person);
+        assertNotNull(savedPerson);
+        assertEquals("DeltaCacheUserName", savedPerson.getUsername());
+
+        store.pushBlocking();
+        Person downloadedPerson = store.pullBlocking(client.query()).getResult().get(0);
+        assertNotNull(downloadedPerson);
+        assertEquals("DeltaCacheUserName", downloadedPerson.getUsername());
+    }
+
+    @Test
+    public void testRead() throws InterruptedException, IOException {
+        client.setUseDeltaCache(true);
+        store = DataStore.collection(Person.COLLECTION, Person.class, StoreType.SYNC, client);
+        assertTrue(store.isDeltaSetCachingEnabled());
+        testManager.cleanBackend(store, StoreType.SYNC);
+
+        Person person = new Person();
+        person.setUsername("DeltaCacheUserName");
+        person.set("CustomField", "CustomValue");
+        Person savedPerson = store.save(person);
+        assertNotNull(savedPerson);
+        assertEquals("DeltaCacheUserName", savedPerson.getUsername());
+
+        store.pushBlocking();
+        Person downloadedPerson = store.pullBlocking(client.query()).getResult().get(0);
+        assertNotNull(downloadedPerson);
+        assertEquals("DeltaCacheUserName", downloadedPerson.getUsername());
+
+        List<Person> personList = testManager.find(store, client.query()).getResult();
+        Person person1 = personList.get(0);
+        assertNotNull(person1);
+        assertNotNull(person1.getUsername());
+        assertEquals("DeltaCacheUserName", person1.getUsername());
+    }
+
+    @Test
+    public void testReadALotOfItems() throws InterruptedException, IOException {
+        client.setUseDeltaCache(true);
+        store = DataStore.collection(Person.COLLECTION, Person.class, StoreType.SYNC, client);
+        assertTrue(store.isDeltaSetCachingEnabled());
+        testManager.cleanBackend(store, StoreType.SYNC);
+
+        for (int i = 0; i < 10; i++) {
+            Person person = new Person();
+            person.setUsername("DeltaCacheUserName_" + i);
+            Person savedPerson = store.save(person);
+            assertNotNull(savedPerson);
+            assertEquals("DeltaCacheUserName_" + i, savedPerson.getUsername());
+        }
+
+        store.pushBlocking();
+        List<Person> pulledPersons = store.pullBlocking(client.query()).getResult();
+        assertNotNull(pulledPersons);
+        assertEquals(10, pulledPersons.size());
+
+        Person person;
+        for (int i = 0; i < 10; i++) {
+            person = pulledPersons.get(i);
+            assertEquals("DeltaCacheUserName_" + i, person.getUsername());
+        }
+
+        pulledPersons = store.pullBlocking(client.query()).getResult();
```

Comment:
One thing you could do is use a separate store of type `NETWORK` to make changes on the backend (e.g. add a new entity, update an existing one). Then your subsequent pull can verify that the addition or update is reflected in the results.

Reply:
Added